klog.go

```diff
@@ -823,19 +828,70 @@ func kvListFormat(b *bytes.Buffer, keysAndValues ...interface{}) {
 			v = missingValue
 		}
 		b.WriteByte(' ')
+		if k, ok := k.(string); ok {
+			// Avoid one allocation when the key is a string, which
+			// normally it should be.
+			b.WriteString(k)
+		} else {
+			b.WriteString(fmt.Sprintf("%s", k))
+		}
 
-		switch v.(type) {
-		case string, error:
-			b.WriteString(fmt.Sprintf("%s=%q", k, v))
+		// The type checks are sorted so that more frequently used ones
+		// come first because that is then faster in the common
+		// cases. In Kubernetes, ObjectRef (a Stringer) is more common
+		// than plain strings
+		// (https://github.com/kubernetes/kubernetes/pull/106594#issuecomment-975526235).
+		switch v := v.(type) {
+		case fmt.Stringer:
+			writeStringValue(b, true, v.String())
+		case string:
+			writeStringValue(b, true, v)
+		case error:
+			writeStringValue(b, true, v.Error())
 		case []byte:
-			b.WriteString(fmt.Sprintf("%s=%+q", k, v))
+			// We cannot use the simpler strconv.Quote here
```

Comment:
This doesn't seem correct - should we fix it?

Reply:
According to the PR which introduced the corresponding test case, `%+q` was intentionally used to handle non-printable characters: https://github.com/kubernetes/klog/pull/237#issuecomment-837615444

Reply:
I don't have any background about why it was decided to handle bytes that way. But because I don't know, I also don't want to change established behavior in this PR.

@serathius, @dims: do you remember?